keras_nlp/layers/start_end_packer.py

```diff
@@ -111,21 +112,23 @@ def call(self, inputs):
         input_is_dense = isinstance(inputs, tf.Tensor)
         input_is_ragged = isinstance(inputs, tf.RaggedTensor)
 
+        input_is_1d = inputs.shape.rank == 1
         if input_is_dense:
-            if inputs.shape.rank != 1:
+            if inputs.shape.rank > 2:
```

Comment:
We can just move the check for rank outside all of these if blocks, apply the same check and error (rank should be either 1 or 2), to both dense and ragged.

Overall flow can be something like

```
if rank < 1 or > 2:
  raise ValueError("msg")
if rank == 1:
  inputs = expand_dims(inputs, axis=0)
if isinstance(inputs, tf.Tensor):
  inputs = tf.RaggedTensor.from_tensor(inputs)
```